### ISSUE TITLE
T2I works ! - Is it really relevant ?

### ISSUE THREAD

**Issue by BreziTasbi** [ivadomed/Counterfactual_synthesis_for_lesion_segmentation#6]
I recently trained again a T2I adapter to force de diffusion model to follow a SC segmentation. For this i trained with canproco and it's "seg-manual" and tested with 20 sct samples and their "sct_deepseg contrast-angnostic" segmentation.
I launched two different trainings. One with the T2I structure straight from the paper and an other one with an extra"zero convolution layer". Inspired by the paper [Control-Net](https://arxiv.org/abs/2302.05543), it nullifies the T2I model influence at the beginning of its training so we can expect it to actually increase the fidelity to input instead of just recovering the original quality. Both training can be visualized on the following :

![Training performances](https://github.com/ivadomed/Counterfactual_synthesis_for_lesion_segmentation/assets/164916503/902c1a7b-fa82-448e-8011-ad9a86d8f26b)

We can witness that both achieve the exact same performance i seem to learn roughly at the same speed. So the zero convolution layer looks useless and have been dropped then.

Nevertheless the T2I training did work on this validation dataset. And looks helpful to segment lesion on sci. Indeed, we can increase the input noise and get a perfectly shaped SC in our counter-factual ! The performance across sci-paris looks better : it brings more sample to a satisfying segmentation F1 score ! (still waaaaay bellow sct dedicated model)

![0807_img2img_segmentation_sci-paris_1_d_0 8_th_0 4_f_filter_T2I_True](https://github.com/ivadomed/Counterfactual_synthesis_for_lesion_segmentation/assets/164916503/eee17f58-6eaa-4f05-babe-5030de9d0b64)

![0607_violin_plot](https://github.com/ivadomed/Counterfactual_synthesis_for_lesion_segmentation/assets/164916503/4f9aa0d1-5838-44a2-8ba1-a13b85c7ed34)

At this point, T2I just looked like going forward to me. But then I looked at the images with T2I input displayed in yellow on the output of the diffusion model i just wondered : do it performs any better than just thresholding distance to the median pixel-value inside the SC ?? Am I not only over-engineering a very very simple unsupervised method ??

**Comment by jcohenadad** [comment]
> does it perform any better than just thresholding distance to the median pixel-value inside the SC?

good question indeed. Baseline comparison is required if we want to move forward with a publication

**Comment by BreziTasbi** [comment]
Just as I expected, a "very very simple unsupervised method" just did better. 
It consist in :
-using "contrast-agnostic" to segment the spinal-coord 
-selecting only voxels above a manually selected threshold in the selected area.

![1007_violin_plot](https://github.com/ivadomed/Counterfactual_synthesis_for_lesion_segmentation/assets/164916503/23e94b58-7edb-4888-aaa8-0e84bbfc5908)

It basically struggles on the same cases but perfoms better overall.

Indeed, since the the SC is basically a uniformed colored area, there is not way more to do than filling up with the correct color for the diffusion model. In particular since It uses a SC segmentation to help him with getting the shape right. 

So basically counterfactual synthesis is irrelevant for segmenting lesions in uniform areas.
